### PR TITLE
Implement some stuff needed for AIGroup

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/common/rts/specialpower.cpp [hunk 1]
```diff
@@ -66,3 +66,12 @@ const SpecialPowerTemplate *SpecialPowerTemplate::Get_FO() const
...
     return static_cast<const SpecialPowerTemplate *>(Friend_Get_Final_Override());
 }
+
+const SpecialPowerTemplate *SpecialPowerStore::Find_Special_Power_Template_By_ID(unsigned int id)
+{
+    for (size_t i = 0; i < m_specialPowerTemplates.size(); i++) {
+        if (m_specialPowerTemplates[i]->Get_ID() == id) {
+            return m_specialPowerTemplates[i];
+        }
+    }
```
Comment (xezon) [inline src/game/common/rts/specialpower.cpp:76]: return nullptr

File: src/game/logic/ai/aigroup.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+#ifdef GAME_DLL
+    Call_Method<void, AIGroup, Object *>(PICK_ADDRESS(0x0054FBF0, 0x008D287E), this, obj);
+#endif
+}
+
+const std::vector<ObjectID> &AIGroup::Get_All_IDs() const
+{
+#ifdef GAME_DLL
+    return Call_Method<const std::vector<ObjectID> &, const AIGroup>(PICK_ADDRESS(0x0054FAC0, 0x008D26A1), this);
```
Comment (xezon) [inline src/game/logic/ai/aigroup.cpp:34]: const AIGroup * ?

File: src/game/logic/ai/aigroup.h [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    GroupID Get_Group_ID();
+    float Get_Speed();
+    const std::vector<ObjectID> &Get_All_IDs() const;
+    bool Get_Center(Coord3D *center);
+    bool Get_Min_Max_And_Center(Coord2D *min, Coord2D *max, Coord3D *center);
+    unsigned int Get_Count();
+    AttitudeType Get_Attitude() const;
+    Object *Get_Special_Power_Source_Object(unsigned int special_power_id);
+    Object *Get_Command_Button_Source_Object(GUICommand command_type);
```
Comment (xezon) [inline src/game/logic/ai/aigroup.h:45]: Get and Is methods read as if they can be const.

File: src/game/logic/object/update/aiupdate.cpp [hunk 4]
```diff
@@ -217,3 +224,10 @@ void AIUpdateInterface::Transfer_Attack(ObjectID old_victim_id, ObjectID new_vic
         PICK_ADDRESS(0x005D5BC0, 0x007FB3A1), this, old_victim_id, new_victim_id);
 #endif
 }
+
+bool AIUpdateInterface::Can_Auto_Acquire_While_Stealthed() const
+{
+    return Get_Object() != nullptr && Get_Object()->Get_Stealth_Update() != nullptr
+        && Get_Object()->Get_Stealth_Update()->Get_Granted_By_Special_Power()
+        || (Get_AI_Update_Module_Data()->m_autoAcquireEnemiesWhenIdle & 2) != 0;
```
Comment (xezon) [inline src/game/logic/object/update/aiupdate.cpp:232]: We have no known enum value for this?

File: src/game/logic/object/update/aiupdate.h [hunk 5]
```diff
@@ -260,6 +263,294 @@ class AICommandInterface
...
+
+    void AI_Attack_Follow_Waypoint_Path_As_Team(const Waypoint *way, int max_shots_to_fire, CommandSourceType cmd_source)
+    {
+        AICommandParms params(AICMD_ATTACKFOLLOW_WAYPOINT_PATH_AS_TEAM, cmd_source);
+        params.m_waypoint = way;
+        params.m_intValue = max_shots_to_fire;
+        AI_Do_Command(&params);
+    }
+
+    void Ai_Attack_Team(const Team *team, int max_shots_to_fire, CommandSourceType cmd_source)
```
Comment (xezon) [inline src/game/logic/object/update/aiupdate.h]: AI_Attack_Team ?

File: src/hooker/setuphooks_zh.cpp [hunk 6]
```diff
@@ -17,6 +17,7 @@
 #endif
 
 #include "aabtree.h"
+#include "ai.h"
```
Comment (xezon) [inline src/hooker/setuphooks_zh.cpp:20]: No function hooks for this change intended? Would the new code take any affect in game?

File: src/game/common/rts/specialpower.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/common/rts/specialpower.cpp:76]: fixed.

File: src/game/logic/ai/aigroup.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/logic/ai/aigroup.cpp:34]: No, this is correct. It wouldn't compile if it was "const AIGroup *"

File: src/game/logic/ai/aigroup.h (hunk 3 above)
Comment (jonwil) [reply src/game/logic/ai/aigroup.h:45]: All the ones that are const in the original are const in here.

File: src/game/logic/object/update/aiupdate.cpp (hunk 4 above)
Comment (jonwil) [reply src/game/logic/object/update/aiupdate.cpp:232]: We do not.

File: src/game/logic/object/update/aiupdate.h (hunk 5 above)
Comment (jonwil) [reply src/game/logic/object/update/aiupdate.h]: fixed.

File: src/hooker/setuphooks_zh.cpp (hunk 6 above)
Comment (jonwil) [reply src/hooker/setuphooks_zh.cpp:20]: This code will be used in a future PR. None of this code will have any effect yet.

**Review by xezon — APPROVED** [review]
Shell map looks as usual. GLA01 mission played half way. Looks ok.

**Pull request merged**